### ISSUE TITLE
HAP-NodeJS stopped working in Raspberry Pi

### ISSUE THREAD

**Issue by rodorgas** [KhaosT/HAP-NodeJS#363]
I was running HAP-NodeJS for weeks on a Raspberry Pi 3. Suddenly, my accessories became unresponsive in Home app. I've removed them from my iPhone, then I couldn't add them back. I've reinstalled node, npm, and HAP-NodeJS.

Demo accessories included in a fresh git clone of HAP can't be added either. They doesn't show up when I try to add an accessory. It works fine when I start the server on my notebook, HAP just doesn't work on this Raspberry Pi and stopped working without any intervention. I've tried to find the accessories using other iPhones, no dice.

Here's the output from the HAP-NodeJS server running in Raspberry Pi 3
```
root@raspberrypi:/home/pi/HAP-NodeJS# DEBUG=* node BridgedCore.js 
*** WARNING *** The program 'nodejs' uses the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=nodejs>
*** WARNING *** The program 'nodejs' called 'DNSServiceRegister()' which is not supported (or only supported partially) in the Apple Bonjour compatibility layer of Avahi.
*** WARNING *** Please fix your application to use the native API of Avahi!
*** WARNING *** For more information see <http://0pointer.de/avahi-compat?s=libdns_sd&e=nodejs&f=DNSServiceRegister>
HAP-NodeJS starting...
Parsing accessory: Fan_accessory.js
Parsing accessory: GarageDoorOpener_accessory.js
Parsing accessory: Light_accessory.js
Parsing accessory: Lock_accessory.js
Parsing accessory: MotionSensor_accessory.js
Parsing accessory: Outlet_accessory.js
Parsing accessory: TemperatureSensor_accessory.js
Parsing accessory: Thermostat_accessory.js
  EventedHTTPServer Server listening on port 51826 +0ms
  EventedHTTPServer [::ffff:192.168.1.37] New connection from client +3m
  EventedHTTPServer [::ffff:192.168.1.37] HTTP server listening on port 35759 +4ms
  EventedHTTPServer [::ffff:192.168.1.37] Client connection closed +27ms
  EventedHTTPServer [::ffff:192.168.1.37] HTTP connection was closed +10ms
  EventedHTTPServer [::ffff:192.168.1.37] HTTP server was closed +6ms
```

All my devices are on the same wifi, and I can test from my notebook the port used by HAP in raspberry is open:
```
➜  HAP-NodeJS git:(master) ✗ nc -zv 192.168.1.20 51826

found 0 associations
found 1 connections:
     1:	flags=82<CONNECTED,PREFERRED>
	outif en0
	src 192.168.1.37 port 61007
	dst 192.168.1.20 port 51826
	rank info not available
	TCP aux info available

Connection to 192.168.1.20 port 51826 [tcp/*] succeeded!
```

I appreciate any ideas to get this working again or how to conduct further investigation.

**Comment by Shifter147** [comment]
I've also encountered this issue lately. 
The accessories parse correct but are not visible on my iPhone. No error messages or weird things are showing on de Raspberry. 

Hope someone can find a solution for this :)

**Comment by bluntiless** [comment]
Hi Rodorgas
Have you tried deleting the persist file and changing the Usernames and pincodes in he accessory.js files? That has worked for me in the past.

**Comment by Shifter147** [comment]
@bluntiless

Thanks, removing the items in persist folder made is possible to add accessories again. (maybe caused by playing with accessories a lot)

But this forced me to re-add my already paired accessories and setup al my scenes containing those accessories again because the all were listed as 'non responsive'.  
Is there a way to prevent this and to keep the paired accessories without adding them again and setting up all my scenes? 

(sorry for highjacking the topic, but i think it's helpful for this problem. )

**Comment by rodorgas** [comment]
I've removed persist folder but it didn't fixed the problem. I even have cloned the repo again (it comes without persist folder) and also changed pins and username at the time.

But I tried again now and it's working fine without any modification, as mysteriously as when it stopped.

**Comment by TeraJaQ** [comment]
i've got the same issue few time ago, but at the end the problem was the memory card on the raspberry pi. I've reinstall all and until now it all ok. i've scheduled a backup of this memory one a week to another SD....

**Comment by rodorgas** [comment]
I hit this issue from time to time. Last time, Pi was shut down by a power outage and HAP didn't came back. I've spent the day digging on this problem, trying to make HAP work on my Raspberry Pi again. I found that this issue is kind of recurring: #384 #444 #261 etc.

The sintoms are weird: even running with `DEBUG=* node Core.js` (or `BridgedCore.js`), I get absolutely no error, everything seems normal. I can hit the port on another computer with `nc -zv 192.168.1.20 20 51826`, and the HAP-NodeJS echoes a message saying it was received. But nothing shows up on iPhone when I try to add an accessory.

What I've done and didn't worked (each item was tried *several* times):
1. Rebooted Raspberry and iPhone (even try with other iPhone)
2. Change `lock.pincode` and `lock.username`
3. Removed `persist/` and `node_modules/`, fresh `npm rebuild` and `npm install`
4. Fresh clone of this repo
5. Complete reinstall of `node` and `npm`
6. Tried several versions of `node`
7. Reinstall `libnss-mdns` and `libavahi-compat-libdnssd-dev`
